src/braket/circuits/noise_helpers.py

```diff
@@ -0,0 +1,286 @@
+
+        target_qubits = QubitSet(target_qubits)
+
+    return target_qubits
+
+
+def apply_noise_to_moments(
+    circuit: Circuit, noise: Iterable[Type[Noise]], target_qubits: QubitSet, position: str
+) -> Circuit:
+    """Apply initialization/readout noise to the circuit.
+
+    When `noise.qubit_count` == 1, `noise` is added to all qubits in `target_qubits`.
+
+    When `noise.qubit_count` > 1, `noise.qubit_count` must be the same as the length of
+    `target_qubits`.
+
+    Args:
+        circuit (Circuit): A ciruit where `noise` is applied to.
+        noise (Iterable[Type[Noise]]): Noise channel(s) to be applied
+        to the circuit.
+        target_qubits (QubitSet): Index or indices of qubits. `noise` is applied to.
+
+    Returns:
+        Circuit: modified circuit.
+    """
+    noise_instructions = []
+    for noise_channel in noise:
+        if noise_channel.qubit_count == 1:
+            new = [Instruction(noise_channel, qubit) for qubit in target_qubits]
+            noise_instructions = noise_instructions + new
+        else:
+            noise_instructions.append(Instruction(noise_channel, target_qubits))
+
+    new_moments = Moments()
+
+    if position == "initialization":
+        for noise in noise_instructions:
+            new_moments.add_noise(noise, "initialization_noise")
+
+    # add existing instructions
+    for moment_key in circuit.moments:
+        instruction = circuit.moments[moment_key]
+        # if the instruction is noise instruction
+        if isinstance(instruction.operator, Noise):
+            new_moments.add_noise(instruction, moment_key.moment_type, moment_key.noise_index)
+        # if the instruction is a gate instruction
+        else:
+            new_moments.add([instruction], moment_key.noise_index)
+
+    if position == "readout":
+        for noise in noise_instructions:
+            new_moments.add_noise(noise, "readout_noise")
+
+    circuit._moments = new_moments
+
+    return circuit
+
+
+def apply_noise_to_gates_helper(
+    noise: Iterable[Type[Noise]],
+    target_qubits: QubitSet,
+    instruction: Instruction,
+    noise_index: int,
+    intersection: QubitSet,
+    noise_applied: bool,
+    new_noise_instruction: Iterable,
+):
+    """Helper function to work out the noise instructions to be attached to a gate.
+
+    Args:
+        noise (Iterable[Type[Noise]]): Noise channel(s) to be applied
+        to the circuit.
+        target_qubits (QubitSet): Index or indices of qubits which `noise` is applied to.
+        instruction (Instruction): Instruction of the gate which `noise` is applied to.
+        noise_index (int): The number of noise channels applied to the gate.
+        intersection (QubitSet): Intersection of target_qubits and the qubits associated
+        with the gate.
+        noise_applied (bool): Whether noise is applied or not.
+        new_noise_instruction (Iterable): current new noise instructions to be attached
+        to the circuit.
```

Comment:
You'll need to add an extra indent for all lines after the first, for each argument